### PR TITLE
DISPATCH-1496: Reset link variables after the drain is complete. Also…

### PR DESCRIPTION
**Pull request by ganeshmurthy**
… removed unused drain flag and related functionality

### CODE REVIEW

**Comment by ganeshmurthy** [comment]
@gemmellr can you please test this and let me know if this works ? It fixes the additional credit problem and passes all existing drain tests in the test suite.
I will write a test for this tomorrow.

@ted-ross please take a look as well.

**Comment by ganeshmurthy** [comment]
I added a test to verify the fix

**Comment by gemmellr** [comment]
No idea on the code changes, but I gave it a try out and all seemed well, the router no longer granted extra credit itself.

File: src/router_node.c [hunk 1]
```diff
@@ -1531,8 +1531,11 @@ static void CORE_link_drained(void *context, qdr_link_t *link)
 
     pn_link_t *plink = qd_link_pn(qlink);
 
-    if (plink)
+    if (plink) {
         pn_link_drained(plink);
+        link->drain_mode = false;
+        link->credit_to_core = 0;
```
Comment (ted-ross) [inline src/router_node.c]: Are these operations safe to do here?  This function is invoked on an IO thread.

File: include/qpid/dispatch/router_core.h [hunk 2]
```diff
@@ -674,6 +674,15 @@ int qdr_link_process_deliveries(qdr_core_t *core, qdr_link_t *link, int credit);
...
 void qdr_link_flow(qdr_core_t *core, qdr_link_t *link, int credit, bool drain_mode);
 
+/**
+ * The passed in link has been drained, it is no longer in drain mode. Sets the link's drain flag to false and sets credit to core to zero.
+ * Call this right after calling pn_link_drained
+ *
+ * @param core - router core
+ * @param link - the link that has been drained
+ */
+void qdr_link_drained(qdr_core_t *core, qdr_link_t *link);
```
Comment (ssorj) [inline include/qpid/dispatch/router_core.h]: I know this is the naming Proton has, but it's confusing.  'qdr_link_drained' looks like a predicate.  Something like 'qdr_link_mark_drained' or '_set_drained' would be a lot clearer IMO.

Comment (ganeshmurthy) [reply include/qpid/dispatch/router_core.h]: I will change this to qdr_link_set_drained. Thanks.

**Pull request closed without merging**